File: mobile/test/pressure_unit_test.dart
import 'package:aetherride_mobile/domain/bike.dart';
import 'package:aetherride_mobile/domain/garage/pressure_unit.dart';
import 'package:flutter_test/flutter_test.dart';

void main() {
  test('Alltag, Gravel, Rennrad merken Druck in bar', () {
    expect(pressureUsesBar(BikeCategory.urban), isTrue);
    expect(pressureUsesBar(BikeCategory.cargo), isTrue);
    expect(pressureUsesBar(BikeCategory.road), isTrue);
    expect(pressureUsesBar(BikeCategory.gravel), isTrue);
    expect(pressureUnitLabel(BikeCategory.folding), 'bar');
  });

  test('MTB merkt Druck in psi', () {
    expect(pressureUsesBar(BikeCategory.mtbAm), isFalse);
    expect(pressureUsesBar(BikeCategory.emtb), isFalse);
    expect(pressureUnitLabel(BikeCategory.dh), 'psi');
    expect(enteredPressureToPsi(22, BikeCategory.mtbAm), 22);
  });

  test('bar-Eingabe landet als psi im Speicher', () {
    final psi = enteredPressureToPsi(4.5, BikeCategory.urban);
    expect(psi, closeTo(65.3, 0.1));
    expect(psiToBar(psi), 4.5);
  });
}

File: src/lib/garage/pressureUnit.test.ts
/**
 * Druck-Einheit — Ausführen: npx tsx src/lib/garage/pressureUnit.test.ts
 */
import {
  barToPsi,
  enteredPressureToPsi,
  formatLoggedTirePressure,
  pressureUnitLabel,
  pressureUsesBar,
  psiToBar,
} from "./pressureUnit";

function assert(cond: boolean, msg: string) {
  if (!cond) throw new Error(msg);
}

assert(pressureUsesBar("urban"), "city bar");
assert(pressureUsesBar("road"), "road bar");
assert(pressureUsesBar("gravel"), "gravel bar");
assert(pressureUsesBar("cargo"), "cargo bar");
assert(pressureUsesBar("folding"), "folding bar");
assert(pressureUsesBar("kids"), "kids bar");
assert(!pressureUsesBar("mtb_am"), "mtb psi");
assert(!pressureUsesBar("emtb"), "emtb psi");
assert(pressureUnitLabel("urban") === "bar", "label bar");
assert(pressureUnitLabel("dh") === "psi", "label psi");

const stored = enteredPressureToPsi(4.5, "urban");
assert(stored > 65 && stored < 66, `4.5 bar → ${stored} psi`);
assert(psiToBar(stored) === 4.5, "round-trip bar");
assert(enteredPressureToPsi(22, "mtb_am") === 22, "mtb passthrough");
assert(barToPsi(1) > 14.4 && barToPsi(1) < 14.6, "1 bar");

const logged = formatLoggedTirePressure(
  [
    {
      values: [
        { adjusterKey: "tire_front.pressure_psi", valueNum: 26.1 },
        { adjusterKey: "tire_rear.pressure_psi", valueNum: 29 },
      ],
    },
  ],
  false
);
assert(logged === "26 / 29 psi", "mtb pressure pair");
assert(
  formatLoggedTirePressure(
    [{ values: [{ adjusterKey: "tire_front.pressure_psi", valueNum: 26.1 }] }],
    true
  ) === "1.8 bar",
  "single bar"
);
assert(formatLoggedTirePressure([], true) === null, "empty setups");

console.log("pressureUnit.test.ts ok");
